packages/artifact/src/internal/utils.ts

```diff
@@ -172,44 +177,57 @@ export function getDownloadRequestOptions(
  * @param {number} uncompressedLength the original size of the content if something is being uploaded that has been compressed
  * @param {number} contentLength the length of the content that is being uploaded
  * @param {string} contentRange the range of the content that is being uploaded
- * @returns appropriate request options to make a specific http call during artifact upload
+ * @returns appropriate request headers to make a specific http call during artifact upload
  */
-export function getUploadRequestOptions(
+export function getUploadHeaders(
   contentType: string,
   isKeepAlive?: boolean,
   isGzip?: boolean,
   uncompressedLength?: number,
   contentLength?: number,
   contentRange?: string
 ): IHeaders {
-  const requestOptions: IHeaders = {}
-  requestOptions['Accept'] = `application/json;api-version=${getApiVersion()}`
+  const headers: IHeaders = {}
+  headers['Accept'] = `application/json;api-version=${getApiVersion()}`
   if (contentType) {
-    requestOptions['Content-Type'] = contentType
+    headers['Content-Type'] = contentType
   }
   if (isKeepAlive) {
-    requestOptions['Connection'] = 'Keep-Alive'
+    headers['Connection'] = 'Keep-Alive'
     // keep alive for at least 10 seconds before closing the connection
-    requestOptions['Keep-Alive'] = '10'
+    headers['Keep-Alive'] = '10'
   }
   if (isGzip) {
-    requestOptions['Content-Encoding'] = 'gzip'
-    requestOptions['x-tfs-filelength'] = uncompressedLength
+    headers['Content-Encoding'] = 'gzip'
+    headers['x-tfs-filelength'] = uncompressedLength
   }
   if (contentLength) {
-    requestOptions['Content-Length'] = contentLength
+    headers['Content-Length'] = contentLength
   }
   if (contentRange) {
-    requestOptions['Content-Range'] = contentRange
+    headers['Content-Range'] = contentRange
   }
 
-  return requestOptions
+  return headers
 }
 
+/**
+ * Creates a new http client that is used by the http-managers when making calls to either upload or download an artifact
+ */
 export function createHttpClient(): HttpClient {
-  return new HttpClient('action/artifact', [
-    new BearerCredentialHandler(getRuntimeToken())
-  ])
+  const requestOptions: IRequestOptions = {
+    // headers get set individually before each call as they can vary significantly
+    headers: [],
+    // keep alive is configured at the http-client level and is used by each client when making calls, this is independent
+    // of the keep-alive header that is used to let the remove server know how the connection should be treated
+    keepAlive: true
```

Comment:
This settings needs to be set and it's independent of the `keep-alive` header

Reply:
This setting makes the http-client save some `agent` behind that scenes that actually makes the calls: https://github.com/actions/http-client/blob/f6aae3dda4f4c9dc0b49737b36007330f78fd53a/index.ts#L143